src/plugins/orangehrmOpenidAuthenticationPlugin/Service/SocialMediaAuthenticationService.php

```diff
@@ -58,26 +91,53 @@ public function initiateAuthentication(AuthProviderExtraDetails $provider, strin
     }
 
     /**
-     * @param OpenIDConnectClient $oidcClient
-     * @throws OpenIDConnectClientException
+     * @return string
      */
-    public function handleCallback(OpenIDConnectClient $oidcClient): void
+    public function getRedirectURL(): string
     {
-        $oidcClient->authenticate();
-
-        try {
-            $isAuthenticated = $oidcClient->authenticate();
-            if ($isAuthenticated) {
-                $credentials = new UserCredential($oidcClient->requestUserInfo('email'));
-                $this->authenticateUser($credentials);
-            }
-        } catch (OpenIDConnectClientException $e) {
-            throw $e;
-        }
+        /** @var UrlGenerator $urlGenerator */
+        $urlGenerator = $this->getContainer()->get(Services::URL_GENERATOR);
+        return $urlGenerator->generate('auth_oidc_login_redirect', [], UrlGenerator::ABSOLUTE_URL);
     }
 
-    private function authenticateUser(UserCredential $userCredential): void
+    /**
+     * @return string
+     */
+    public function getScope(): string
+    {
+        return self::SCOPE;
+    }
+
+    /**
+     * @param UserCredential $userCredential
+     * @return array
+     */
+    public function getOIDCUser(UserCredential $userCredential): array
+    {
+        $userSearchFilterParams = new UserSearchFilterParams();
+        $userSearchFilterParams->setUsername($userCredential->getUsername());
+
+        return $this->getUserDao()->searchSystemUsers($userSearchFilterParams);
+    }
+
+    /**
+     * @param User $user
+     * @param OpenIdProvider $provider
```

Comment:
incomplete doc comment
```suggestion
     * @param OpenIdProvider $provider
     * @return OpenIdUserIdentity
```